File: WebApiRest/Models/Noticia.cs
namespace WebApiRest.Models
{
    public class Noticia
    {
        public Guid IdNoticia { get; set; }
        public string Titular { get; set; }
        public string Descripcion { get; set; }
        public string Url { get; set; }
        public string Imagen { get; set; }
        public int Estado { get; set; }
        public Guid IdCategoria { get; set; }
        public string Categoria { get; set; }
        public DateTime FechaPublicacion { get; set; }
        public int TotalLikes { get; set; }
        public int TotalComents { get; set; }
        public DateTime FechaCreacion { get; set; }
        public DateTime FechaModificacion { get; set; }
    }

    public class NoticiaList
    {
        public string Info { get; set; }
        public int Error { get; set; }
        public List<Noticia> Lista { get; set; }
    }

    // TABLA INTERMEDIA
    public class Usuario_Noticia
    {
        public Guid IdUsuario { get; set; }
        public string Usuario { get; set; }
        public Guid IdNoticia { get; set; }
        public int Likes { get; set; }
        public string Comentario { get; set; }
        public DateTime FechaCreacion { get; set; }
        public DateTime FechaModificacion { get; set; }
    }

    public class Usuario_NoticiaList
    {
        public string Info { get; set; }
        public int Error { get; set; }
        public List<Usuario_Noticia> Lista { get; set; }
    }
}

File: FrontWebApp/src/app/Models/Noticia.ts
export interface Noticia {
  idNoticia: string;
  titular: string;
  descripcion: string;
  url: string;
  imagen: string;
  idCategoria: string;
  categoria: string;
  fechaPublicacion: Date;
  totalLikes: number;
  totalComents: number;
}

export interface Reaccion {
  idNoticia: string;
  idUsuario: string;
  comentario: string;
  likes: number;
  usuario: string;
}

export interface Noticia_Reaccion {
  noticia: Noticia;
  userLiked: number;
  comentarioList: Reaccion[];
}

export interface Usuario_Noticia {
  idUsuario: string;
  idNoticia: string;
  likes: number;
  comentario: string;
}
